storage/scylla_api.c

```diff
@@ -0,0 +1,430 @@
+      get_blob(row, &(((*transaction) + idx)->bundle), "bundle");
+      get_blob(row, &(((*transaction) + idx)->address), "address");
+      get_blob(row, &(((*transaction) + idx)->transaction), "transaction");
+      get_blob(row, &(((*transaction) + idx)->timestamp), "timestamp");
+      get_text(row, &(((*transaction) + idx)->message), "message");
+      get_blob(row, &(((*transaction) + idx)->trunk), "trunk");
+      get_blob(row, &(((*transaction) + idx)->branch), "branch");
+      cass_value_get_int64(cass_row_get_column_by_name(row, "value"), &(((*transaction) + idx)->value));
+      idx++;
+    }
+
+    cass_result_free(result);
+    cass_iterator_free(iterator);
+  }
+
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return rc;
+}
+
+status_t select_transactions_with_bundle(CassSession* session, const Select_Way select_way,
+                                         const Select_Where** select_where, const size_t select_num) {
+  Transaction* output;
+  int outputNum = 0;
+  CassStatement* statement = NULL;
+  const CassPrepared* select_prepared = NULL;
+
+  if (prepare_query(session, select_query[select_way].query, &select_prepared) == CASS_OK) {
+    statement = cass_prepared_bind(select_prepared);
+    for (size_t i = 0; i < select_num; i++) {
+      statement = cass_prepared_bind(select_prepared);
+      if (select_way == WITH_BUNDLE) {
+        cass_statement_bind_bytes_by_name(statement, "bundle", ((*select_where) + i)->bundle,
+                                          strlen((char*)((*select_where) + i)->bundle));
+      } else if (select_way == WITH_BUNDLE_AND_ADDRESS) {
+        printf("%lu\n", strlen((char*)((*select_where) + i)->bundle));
+        cass_statement_bind_bytes_by_name(statement, "bundle", ((*select_where) + i)->bundle,
+                                          strlen((char*)((*select_where) + i)->bundle));
+        cass_statement_bind_bytes_by_name(statement, "address", ((*select_where) + i)->address,
+                                          strlen((char*)((*select_where) + i)->address));
+      }
+      select_from_bundleTable(session, statement, &output, &outputNum);
+      /* serialize to output format here*/
+      SCYLLA_LOG_DEBUG("row = %d\n", outputNum);
+      for (int i = 0; i < outputNum; i++) {
+        SCYLLA_LOG_DEBUG("%s %s %s %s %ld %s %s %s\n", output[i].bundle, output[i].address, output[i].transaction,
+                         output[i].message, output[i].value, output[i].timestamp, output[i].trunk, output[i].branch);
+      }
+
+      free(output);
+    }
+    cass_prepared_free(select_prepared);
+  } else {
+    SCYLLA_LOG_ERROR("%s\n", "prepare SELECT query fail");
+    return SC_STORAGE_PREPARE_QUERY_FAIL;
+  }
+  return SC_OK;
+}
+
+status_t scylla_test(char* hosts) {
+  CassCluster* cluster = NULL;
+  CassSession* session = cass_session_new();
+  if (init(cluster, session, hosts) != 0) {
+    return SC_STORAGE_CONNECT_FAIL;
+  }
+
+  Transaction transactions[] = {
```

Comment:
Explain the following items within comments.

Reply:
I moved it to tests/test_scylla.c within some comments.